pkgs/development/python-modules/os-client-config/default.nix

```
@@ -0,0 +1,47 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  glanceclient,
+  oslotest,
+  openstacksdk,
+  stestr,
+  setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "os-client-config";
+  version = "2.1.0";
+  pyproject = true;
+
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-q8OKNR+MAG009+5fP2SN5ePs9kVcxdds/YidKRzfP04=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [
+    glanceclient
+  ];
```

```suggestion
  build-system = [ setuptools ];

  dependencies = [
    glanceclient
  ];
```

We can use build-system and dependencies here.

---

Should I use dependencies for this package or could I keep the one I have pushed ?  
What is the difference and how should I do ?

---

`glanceclient` is a Python runtime dependency, so it should go in `dependencies`

`build-system` and `dependencies` were introduced recently to separate runtime & build time Python dependencies, and solve dependency propagation in Python.

> When should I use dependencies instead of propagatedBuildInputs ? Should, in python packing, propagatedBuildInputs be used ?

If I am not mistaken,

* `build-system` is for build-time Python dependencies only (`setup_requires`)
* `dependencies` is for runtime Python dependencies only (`install_requires`)
* `propagatedBuildInputs` can be used if your package relies on a runtime dependency that is not a Python dependency for a Python package

You can find more documentation/information here:
* https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters
* https://github.com/NixOS/nixpkgs/pull/271597

---

Thanks for the explanation. I'm gonna update the MR then